src/components/ResetPasswordForm.js

```diff
@@ -1,76 +1,152 @@
+              <div className="form-horizontal">
+                <div className="form-group">
+                  <label htmlFor="spEmail" className="col-xs-12 col-sm-4 control-label">Email or Username</label>
+                  <div className="col-xs-12 col-sm-4">
+                    <input className="form-control" id="spEmail" name="email" placeholder="Your Email Address" />
+                  </div>
+                </div>
+                <div className="form-group">
+                  <div className="col-sm-offset-4 col-xs-12">
+                    <p spIf="form.error"><span spBind="form.errorMessage" /></p>
+                    <button type="submit" className="btn btn-primary">Request Password Reset</button>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </ResetPasswordForm>
+    );
+  }
+}
+
+@ReactMixin.decorate(History)
 export default class ResetPasswordForm extends React.Component {
   state = {
     fields: {
       email: ''
     },
-    isRequestSent: false,
-    errorMessage: null
+    errorMessage: null,
+    isFormProcessing: false,
+    isFormSent: false
   };
 
   onFormSubmit(e) {
-    var self = this;
     e.preventDefault();
 
-    UserActions.forgotPassword(this.state.fields, function (err) {
+    var next = (err, data) => {
       if (err) {
-        self.setState({ errorMessage: err.message });
-        return;
+        return this.setState({
+          isFormProcessing: false,
+          errorMessage: err.message
+        });
       }
-      self.setState({ isRequestSent: true });
+
+      // If the user didn't specify any data,
+      // then simply default to what we have in state.
+      data = data || this.state.fields;
+
+      UserActions.forgotPassword(this.state.fields, (err) => {
+        if (err) {
+          this.setState({
+            errorMessage: err.message
+          });
+        } else {
+          this.setState({
+            isFormSent: true
+          });
+        }
+      });
+    };
+
+    this.setState({
+      isFormProcessing: true
     });
+
+    if (this.props.onSubmit) {
+      e.data = this.state.fields;
+      this.props.onSubmit(e, next);
+    } else {
+      next(null, this.state.fields);
+    }
+  }
+
+  _mapFormFieldHandler(element, tryMapField) {
+    if (['input', 'textarea'].indexOf(element.type) > -1) {
+      if (element.props.type !== 'submit') {
+        switch(element.props.name) {
+          case 'email':
+            tryMapField('email');
+            break;
+        }
+      }
+    }
   }
 
-  onEmailChanged(e) {
-    this.state.fields.email = e.target.value;
+  _spIfHandler(action, element) {
+    var test = null;
+
+    switch (action) {
+      case 'form.processing':
+        test = this.state.isFormProcessing;
+        break;
+      case 'form.sent':
+        test = this.state.isFormSent;
+        break;
+      case 'form.error':
+        test = !!this.state.errorMessage;
```

Comment:
Change to `Boolean(this.state.errorMessage)` or `this.state.errorMessage !== null` instead.